third_party/move/move-core/types/src/value.rs

```diff
@@ -691,7 +704,9 @@ impl serde::Serialize for MoveValue {
             MoveValue::U128(i) => serializer.serialize_u128(*i),
             MoveValue::U256(i) => i.serialize(serializer),
             MoveValue::Address(a) => a.serialize(serializer),
-            MoveValue::Signer(a) => a.serialize(serializer),
+            MoveValue::Signer(a) => {
+                MoveStruct::new_variant(0, vec![MoveValue::Address(*a)]).serialize(serializer)
```

Comment:
this deserves a comment on what it does here and why

Reply:
+1, here the comment should say why we construct a master signer, and tag of `0` should really be a somewhere defined constant, or at least (but still with a constant) a constructor like `master()/permissioned()`, etc.

Reply:
yeah, I think it would be better to have utility functions that create and unpack master/permissioned signer - so layout is only in one place.

what I think should be commented here is why are we creating master signer here, and in which format would serialize produce and why (i.e. comments in the code to distinguish where we are using enum serialization vs backward compatible one)